### PR TITLE
Run 'pip check' in tests

### LINKED ISSUE #1: `pip check` fails with pip==24.2: certipy 0.1.3 is not supported on this platform
### Issue

Running a `pip check` in a fresh conda environment with latest pip==24.2 reports a problem that's likely caused by invalid wheel metadata. More info and discussion can be found on this issue:
https://github.com/pypa/pip/issues/11054

To reproduce
```
$ conda create -n test pip=22.4 certipy
$ conda activate test
$ pip check
certipy 0.1.3 is not supported on this platform
```

Note that `pip check` does not fail if the `certipy` package is installed with pip from PYPI.

### PR DESCRIPTION
**Pull request by danielhollas**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #1

### CODE REVIEW

**Comment by danielhollas** [comment]
@conda-forge-admin, please rerender

**Comment by danielhollas** [comment]
@minrk can you take a look? Thanks! 🙏

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ source:
 
 build:
   noarch: python
-  number: 0
+  number: 1
```
Comment (minrk) [inline recipe/meta.yaml:14]: Technically don't need a version bump since this doesn't affect packaging, only tests. But not a problem.

Comment (danielhollas) [reply recipe/meta.yaml:14]: A new build was needed to resolve #1. (I didn't look deep enough to see why the original build was failing the pip check, but evidently just rebuilding was enough to resolve it)

Thank you for merging!

Comment (minrk) [reply recipe/meta.yaml:14]: Ah, of course! I should have read #1 more carefully. Makes perfect sense, thanks!

**Pull request merged**